### ISSUE TITLE
[INSTALL] HDF5 1.12.2 and Netcdf 4.9.0

### ISSUE THREAD

**Issue by junwang-noaa** [NOAA-EMC/hpc-stack#491]
Note: Please search to see if an issue already exists for the software you are requesting to be installed.

**Which software in the stack would you like installed?**
HDF5 and Netcdf
Some other libraries that have dependency on these two libraries need to be rebuilt. E.g. ESMF (8.3.1 preferred) and MAPL(latest version preferred).

**What is the version/tag of the software?**
HDF5 1.12.2
Netcdf 4.9.0 (it needs to be built with HDF5 1.12.2)

**What compilation options would you like set?**
The same as previous options.

**Which machines would you like to have the software installed?**
See currently managed HPC machines.

**Any other relevant information that we should know to correctly install the software??**
Please describe in as much detail as possible.

**Additional context**
Add any other context or screenshots about the install request here.

**Comment by junwang-noaa** [comment]
@GeorgeVandenberghe-NOAA FYI.

**Comment by KateFriedman-NOAA** [comment]
Thanks for opening this @junwang-noaa. I'm going to follow this issue as well.

**Comment by junwang-noaa** [comment]
@climbfuji Can we build the two libraries (and associated libraried) in spack stack?

**Comment by climbfuji** [comment]
> @climbfuji Can we build the two libraries (and associated libraried) in spack stack?

It's too late for this release, the code freeze is today and we haven't tested building and installing it anywhere (and there are still PRs that need to be merged for these new versions). We can do that after the spack-stack-1.1.0 as a follow-up.

**Comment by climbfuji** [comment]
I am recording the next updates for the UFS here: https://github.com/NOAA-EMC/spack-stack/issues/344

**Comment by junwang-noaa** [comment]
@edwardhartnett I want to confirm that the netcdf 4.9.0 has the Bit grooming algorithm that we can use in the ufsi-weather-model write grid component, is it correct? Thanks

**Comment by Hang-Lei-NOAA** [comment]
@junwang-noaa I have added intel installations for the requested sets on Hera hpc-stack offical installations.
The  complete sets:
hdf5 1.12.2
netcdf 4.9.0
netcdf-fortran 4.6.0
netcdf-cxx 4.3.1

You can start from there, then we can add spack-stack soon.

**Comment by jkbk2004** [comment]
@Hang-Lei-NOAA can I make sure ESMF and MAPL are only two libs dependent on netcdf?

**Comment by Hang-Lei-NOAA** [comment]
No, a lot of libs are depends on netcdf. You have to search. e.g. wgrib2,
upp, etc.

**Comment by jkbk2004** [comment]
So fresh rebuild might be short cut for netcdf4.9.0, right?

**Comment by Hang-Lei-NOAA** [comment]
Yes
